src-built-in/components/windowTitleBar/src/windowTitleBarComponent.jsx

```diff
@@ -407,7 +407,15 @@ class WindowTitleBar extends React.Component {
 				</div>
 				<div className={rightWrapperClasses} ref={this.setToolbarRight}>
 					{showDockingIcon ? <DockingButton /> : null}
-					{this.state.alwaysOnTopButton && showMinimizeIcon ? <AlwaysOnTop /> : null}
+					{/** DH 11/22/2019
+					 * Because AlwaysOnTop is a "smart" component that registers
+					 * event handlers, etc., it's not a good idea to constantly mount
+					 * and unmount it. To prevent this, we pass in a "visible" prop that,
+					 * if false, sets "display: none". Ideally, AlwaysOnTop should be a "dumb"
+					 * component, and all the event handlers, etc. should be registered in a parent component,
+					 * as this simplifies the UI and allows React to better optimize under the hood.
+					 */}
+					<AlwaysOnTop visible={this.state.alwaysOnTopButton && showMinimizeIcon}/>
```

Comment:
This is a little concerning. Is there a reason alwaysOnTop can't be a dumb component? Did it work the same way before?

Reply:
Answer is the work is too big to do in this effort.

Reply:
Yes, it worked the same way before, but had an intrinsic race condition. This fixes the race condition, but leaves the (sub-optimal) design in place (we can fix it in the seed refactor)